subsys/net/lib/sockets/socketpair.c

```diff
@@ -191,6 +191,10 @@ static void spair_delete(struct spair *spair)
 	res = k_poll_signal_raise(&spair->writeable, SPAIR_SIG_CANCEL);
 	__ASSERT(res == 0, "k_poll_signal_raise() failed: %d", res);
 
+	if (remote != NULL && have_remote_sem) {
+		k_sem_give(&remote->sem);
+	}
+
 	/* ensure no private information is released to the memory pool */
 	memset(spair, 0, sizeof(*spair));
```

Comment:
What happens now after we have release the sem, could some other code access the memory that is cleared by the following memset()?

@cfriedt About the memset(), if we do not want to leak private information, should we just clear the buf and not the whole spair? Wouldn't there be problems if some other thread is to be about to access the same spair and all the data is cleared?

Reply:
> @cfriedt About the memset(), if we do not want to leak private information, should we just clear the buf and not the whole spair?

Technically speaking, we don't even need to call memset.

Reply:
> What happens now after we have release the sem, could some other code access the memory that is cleared by the following memset()?

From a quick look, the `SPAIR_SIG_CANCEL` doesn't need any sync, before accessing the `struct spair`, the signal is always check and if its cancelled, then `struct spair` isn't accessed, see https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/sockets/socketpair.c#L528.